.github/workflows/circuit-collator-release.yml

```diff
@@ -53,8 +53,43 @@ jobs:
           if-no-files-found: error
           retention-days: 1
 
+  publish-docker-image:
+    runs-on: ubuntu-22.04
+    needs: build-release
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+          token: ${{ secrets.GH_PAT }}
+
+      - name: 📥 Download circuit collator
+        uses: actions/download-artifact@v2
+        with:
+          name: ${{ github.sha }}-collator-release-artifacts
+          path: ./target/release/
+
+      - name: 🏷️ Get the version tag
+        run: echo "PUSHED_TAG=${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
+
+      - name: 🐋 Login to Docker Hub
```

Comment:
As far as im aware, dockerhub rate limits pulls, so if we try to scale, we might fail to scale. Another issue is a would-be attacker could just pull our docker image as a bot and effectively denial of service. The alternative here is ghcr(https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry)

Reply:
your scenario isn't valid here - this pipeline only runs when we push a git tag

Reply:
That's not exactly true. If our git flow is: push image tag > trigger deployment. Then a third party can block that deployment easily as above.